plugins/file/plugin.go

```diff
@@ -125,24 +205,120 @@ func setupFile6(args ...string) (handler.Handler6, error) {
+		}
+		log.Printf("plugins/file: loaded %d leases from %s", len(records), filename)
+		StaticRecords = records
+	} else {
+		if len(args) < 3 {
+			return nil, nil, errors.New("plugins/file: need a file name, server IP, netmask and a lease time")
+		}
+		filename = args[0]
+		if filename == "" {
+			return nil, nil, errors.New("plugins/file: got empty file name")
+		}
+		var err error
+		_, network, err = net.ParseCIDR(args[1])
+		if err != nil {
+			return Handler6, Handler4, errors.New("plugins/file: expected an IPv4 address, got: " + args[1])
+		}
+		LeaseTime, err = time.ParseDuration(args[2])
+		if err != nil {
+			return Handler6, Handler4, errors.New("plugins/file: expected an uint32, got: " + args[2])
+		}
+		records, err := LoadDHCPv4Records(filename)
+		if err != nil {
+			return nil, nil, fmt.Errorf("plugins/file: failed to load DHCPv4 records: %v", err)
+		}
+		DHCPv4Records = records
+
+		rand.Seed(time.Now().Unix())
+
+		log.Printf("plugins/file: loaded %d leases from %s", len(DHCPv4Records), filename)
+	}
+
+	return Handler6, Handler4, nil
+}
+func createIP(network *net.IPNet) (*Record, error) {
+	ip := []byte{random(1, 254), random(1, 254), random(1, 254), random(1, 254)}
+	for i := 0; i < 4; i++ {
+		ip[i] = (ip[i] & (network.Mask[i] ^ 255)) | (network.IP[i] & network.Mask[i])
+	}
+	taken := checkIfTaken(ip)
+	for taken {
+		ipInt := binary.BigEndian.Uint32(ip)
+		ipInt++
+		binary.BigEndian.PutUint32(ip, ipInt)
+		if !network.Contains(ip) {
+			break
+		}
+		taken = checkIfTaken(ip)
+	}
+	for taken {
+		ipInt := binary.BigEndian.Uint32(ip)
+		ipInt--
+		binary.BigEndian.PutUint32(ip, ipInt)
+		if !network.Contains(ip) {
+			return &Record{}, errors.New("plugins/file: no new IP addresses available")
+		}
+		taken = checkIfTaken(ip)
 	}
-	filename := args[0]
-	if filename == "" {
-		return nil, nil, errors.New("plugins/file: got empty file name")
+	return &Record{IP: ip, leaseTime: LeaseTime, leased: time.Now().Unix()}, nil
+
+}
+func random(min int, max int) byte {
+	return byte(rand.Intn(max-min) + min)
+}
+func checkIfTaken(ip net.IP) bool {
+	taken := false
+	if ip.String() == network.IP.String() {
+		return true
 	}
-	records, err := LoadDHCPv6Records(filename)
+	for _, v := range DHCPv4Records {
+		if v.IP.String() == ip.String() && (v.leased+int64(v.leaseTime.Seconds()) > time.Now().Unix()) {
+			taken = true
+			break
+		}
+	}
+	return taken
+}
+func saveIPAddress(record *Record, mac net.HardwareAddr) error {
+	f, err := os.OpenFile(filename, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
 	if err != nil {
-		return nil, nil, fmt.Errorf("plugins/file: failed to load DHCPv6 records: %v", err)
+		return err
 	}
-	log.Printf("plugins/file: loaded %d leases from %s", len(records), filename)
-	StaticRecords = records
+	defer f.Close()
+	_, err = f.WriteString(mac.String() + " " + record.IP.String() + " " + strconv.FormatUint(uint64(record.leaseTime.Seconds()), 10) + "s " + strconv.FormatInt(record.leased, 10) + "\n")
```

Comment:
Same as above, can't this be simplified to contain only the timestamp of when the lease expires instead of when the lease was assigned plus its duration?